### ISSUE TITLE
kubeflow-presubmit error when change/modfiy under tensorflow-notebook-image

### ISSUE THREAD

**Issue by xauthulei** [kubeflow/kubeflow#3995]
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
I have committed a PR #3879, it is failed at "kubeflow-presubmit". 

**What did you expect to happen:**
Fixed it and tested pass.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]
```
INFO|2019-08-23T09:08:40|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| level=info msg="Using context \"gke_kubeflow-ci_us-east1-d_kubeflow-testing\" from kubeconfig file \"/root/.kube/config\"" 

INFO|2019-08-23T09:08:40|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| level=info msg="Creating environment \"kubeflow-presubmit-tf-notebook-release-3879-6cc775a-0800-b726\" with namespace \"kubeflow-test-infra\", pointing to cluster at address \"https://35.196.213.148\"" 

INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| panic: runtime error: invalid memory address or nil pointer dereference 

INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x456452] 

INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| 

INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| goroutine 1 [running]: 

INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| github.com/ksonnet/ksonnet/vendor/github.com/ksonnet/ksonnet-lib/ksonnet-gen/ksonnet.parsePaths(0xc4204d4300, 0x6, 0x1, 0xc) 

INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| /Users/bryan/go/src/github.com/ksonnet/ksonnet/vendor/github.com/ksonnet/ksonnet-lib/ksonnet-gen/ksonnet/paths.go:33 +0x1f1 
```

More details:
```
INFO|2019-08-23T09:08:40|/src/kubeflow/testing/py/kubeflow/testing/run_e2e_workflow.py|204| Triggering workflow tf-notebook-release because components/tensorflow-notebook-image/Dockerfile in dir components/tensorflow-notebook-image/* is modified.
INFO|2019-08-23T09:08:40|/src/kubeflow/testing/py/kubeflow/testing/util.py|45| Running: ks env add kubeflow-presubmit-tf-notebook-release-3879-6cc775a-0800-b726 --namespace=kubeflow-test-infra 
cwd=/src/kubeflow/kubeflow/components/tensorflow-notebook-image/releaser
INFO|2019-08-23T09:08:40|/src/kubeflow/testing/py/kubeflow/testing/util.py|60| Subprocess output:

INFO|2019-08-23T09:08:40|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| level=info msg="Using context \"gke_kubeflow-ci_us-east1-d_kubeflow-testing\" from kubeconfig file \"/root/.kube/config\""
INFO|2019-08-23T09:08:40|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| level=info msg="Creating environment \"kubeflow-presubmit-tf-notebook-release-3879-6cc775a-0800-b726\" with namespace \"kubeflow-test-infra\", pointing to cluster at address \"https://35.196.213.148\""
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| panic: runtime error: invalid memory address or nil pointer dereference
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x456452]
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| 
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| goroutine 1 [running]:
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| github.com/ksonnet/ksonnet/vendor/github.com/ksonnet/ksonnet-lib/ksonnet-gen/ksonnet.parsePaths(0xc4204d4300, 0x6, 0x1, 0xc)
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| /Users/bryan/go/src/github.com/ksonnet/ksonnet/vendor/github.com/ksonnet/ksonnet-lib/ksonnet-gen/ksonnet/paths.go:33 +0x1f1
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| github.com/ksonnet/ksonnet/vendor/github.com/ksonnet/ksonnet-lib/ksonnet-gen/ksonnet.NewCatalog(0xc4204d4300, 0xc42074d0e8, 0x1, 0x1, 0x40, 0x0, 0x0)
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| /Users/bryan/go/src/github.com/ksonnet/ksonnet/vendor/github.com/ksonnet/ksonnet-lib/ksonnet-gen/ksonnet/catalog.go:76 +0x1de
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| github.com/ksonnet/ksonnet/vendor/github.com/ksonnet/ksonnet-lib/ksonnet-gen/ksonnet.GenerateLib(0xc4216a2040, 0xe, 0x0, 0x0, 0x0)
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| /Users/bryan/go/src/github.com/ksonnet/ksonnet/vendor/github.com/ksonnet/ksonnet-lib/ksonnet-gen/ksonnet/ksonnet.go:25 +0xce
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| github.com/ksonnet/ksonnet/pkg/util/kslib.astBuilder(0xc4216a2040, 0xe, 0xc420d00000, 0x3f0dc0, 0x3ffe00, 0x0, 0x0, 0x0)
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/util/kslib/ksonnet.go:87 +0x40
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| github.com/ksonnet/ksonnet/pkg/util/kslib.Ksonnet(0xc420d00000, 0x3f0dc0, 0x3ffe00, 0x0, 0x0, 0x0)
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/util/kslib/ksonnet.go:83 +0x3fd
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| github.com/ksonnet/ksonnet/pkg/lib.(*Manager).GenerateLibData(0xc42139f500, 0x1, 0x1dc69a0, 0x2a15f78)
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/lib/lib.go:83 +0x8a
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| github.com/ksonnet/ksonnet/pkg/app.updateLibData(0x1dc69a0, 0x2a15f78, 0xc4213e4cb0, 0xf, 0xc42142ba90, 0x48, 0x1, 0xc4211435e0, 0x43b4ab, 0xc420039248, ...)
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/app/app.go:128 +0xba
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| github.com/ksonnet/ksonnet/pkg/app.(*App010).AddEnvironment(0xc42000c6e8, 0x7ffd1d0d351e, 0x3d, 0xc4213e4cb0, 0xf, 0xc42143dd40, 0x0, 0x1a4, 0x0)
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/app/app010.go:62 +0x427
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| github.com/ksonnet/ksonnet/pkg/env.(*creator).Create(0xc420160870, 0xc420160870, 0xc420282de0)
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/env/create.go:117 +0x979
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| github.com/ksonnet/ksonnet/pkg/env.Create(0x1dcb1a0, 0xc42000c6e8, 0xc4208f4e20, 0x16, 0x7ffd1d0d3568, 0x13, 0x7ffd1d0d351e, 0x3d, 0xc4213e4cb0, 0xf, ...)
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/env/create.go:42 +0x157
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| github.com/ksonnet/ksonnet/pkg/actions.(*EnvAdd).Run(0xc420282de0, 0xc420282de0, 0x0)
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/actions/env_add.go:71 +0xeb
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| github.com/ksonnet/ksonnet/pkg/actions.RunEnvAdd(0xc421438090, 0xc4208650e0, 0x5)
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/actions/env_add.go:30 +0x5c
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| github.com/ksonnet/ksonnet/pkg/clicmd.runAction(0x5, 0xc421438090, 0x187324a, 0x8)
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/clicmd/actions.go:113 +0x66
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| github.com/ksonnet/ksonnet/pkg/clicmd.glob..func8(0x29d4840, 0xc42081faa0, 0x1, 0x2, 0x0, 0x0)
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| /Users/bryan/go/src/github.com/ksonnet/ksonnet/pkg/clicmd/env_add.go:87 +0x505
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra.(*Command).execute(0x29d4840, 0xc42081fa40, 0x2, 0x2, 0x29d4840, 0xc42081fa40)
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| /Users/bryan/go/src/github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra/command.go:753 +0x468
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x29d96a0, 0x13, 0x18b9332, 0x30)
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| /Users/bryan/go/src/github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra/command.go:843 +0x30a
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra.(*Command).Execute(0x29d96a0, 0xc4200b4058, 0x0)
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| /Users/bryan/go/src/github.com/ksonnet/ksonnet/vendor/github.com/spf13/cobra/command.go:791 +0x2b
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| main.main()
INFO|2019-08-23T09:08:42|/src/kubeflow/testing/py/kubeflow/testing/util.py|71| /Users/bryan/go/src/github.com/ksonnet/ksonnet/cmd/ks/main.go:35 +0x85
Traceback (most recent call last):
  File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main
    "__main__", fname, loader, pkg_name)
  File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
    exec code in run_globals
  File "/src/kubeflow/testing/py/kubeflow/testing/run_e2e_workflow.py", line 418, in <module>
    final_result = main()
  File "/src/kubeflow/testing/py/kubeflow/testing/run_e2e_workflow.py", line 408, in main
    return run(args, file_handler)
  File "/src/kubeflow/testing/py/kubeflow/testing/run_e2e_workflow.py", line 237, in run
    cwd=w.app_dir)
  File "/src/kubeflow/testing/py/kubeflow/testing/util.py", line 87, in run
    " ".join(command), process.returncode), "\n".join(output))
subprocess.CalledProcessError: Command 'cmd: ks env add kubeflow-presubmit-tf-notebook-release-3879-6cc775a-0800-b726 --namespace=kubeflow-test-infra exited with code 2' returned non-zero exit status 2

```
**Environment:**

Env from the log:
ksonnet version: 0.11.0
jsonnet version: v0.10.0

**Comment by xauthulei** [comment]
Reference the error :https://github.com/ksonnet/ksonnet/issues/883
Seems we need to upgrade the ks version. 
If so, we need to change the `apiVersion` to `0.3.0` in the [app.yaml](https://github.com/kubeflow/kubeflow/blob/master/components/tensorflow-notebook-image/releaser/app.yaml) compared with other [app.yaml](https://github.com/kubeflow/kubeflow/blob/master/components/jupyter-web-app/releaser/app.yaml) under `component/jupyter-web-app`, also more details can see: [ks_util.py](https://github.com/kubeflow/testing/blob/master/py/kubeflow/testing/ks_util.py)

**Comment by jlewi** [comment]
@xauthulei thanks for debugging; Do you want to submit a PR to upgrade the tests?

**Comment by xauthulei** [comment]
@jlewi, Yea, I am willing to, Thanks

**Comment by xauthulei** [comment]
@kkasravi, @jlewi , I am try to fix this issue, Could you review the PR #3879 . Thanks.

**Comment by xauthulei** [comment]
As the PR #3879 is merged, so close  this issue.
